src/quick-lint-js/diag/diagnostic-types-2.h

```diff
@@ -3485,6 +3485,15 @@ struct Diag_Unintuitive_Bitshift_Precedence {
   Source_Code_Span bitshift_operator;
   Source_Code_Span and_operator;
 };
+
+struct Diag_Using_Dot_After_Optional_Chaining {
+  [[qljs::diag("E0717", Diagnostic_Severity::warning)]]
+  [[qljs::message("using a '{0}' after a '{1}' might fail, "
+                  "since '{1}' might return 'undefined'.",
```

Comment:
Nit: It will help maintainers and translators if you use substitutions sparingly. (I don't always follow this rule myself...) `{0}` will always be `.`, and `{1}` will always be `?.`.
Nit: Remove the period at the end of the message.
```suggestion
  [[qljs::message("using a '.' after a '?.' might fail, "
                  "since '?.' might return 'undefined'",
```